src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionWithFlavors.cs

```diff
@@ -50,7 +51,7 @@ public override bool HasActionSets
             // Light bulb will always invoke this property on the UI thread.
             AssertIsForeground();
 
-            if (_actionSets == null)
+            if (_actionSets.IsDefault)
```

Comment:
Yes, it has operator for nullable type so original code works fine, but I had same question as Kevin and only found out that once I dug into the ImmutableArray source code. I changed it to IsDefault so that it is more explicit.


-          Heejae

From: Kevin Pilch-Bisson [mailto:notifications@github.com]
Sent: Friday, August 14, 2015 8:08 AM
To: dotnet/roslyn <roslyn@noreply.github.com>
Cc: HeeJae Chang <Heejae.Chang@microsoft.com>
Subject: Re: [roslyn] added explict check for uninitialized immutable array (#4550)


In src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionWithFlavors.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2froslyn%2fpull%2f4550%23discussion_r37085320&data=01%7c01%7cHeejae.Chang%40microsoft.com%7caf7b1b89ec0b4316707208d2a4ba15be%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=c29qPFWumnN0bNLxDr%2bjThCnqGbCAsMXWh7t4a0gbmk%3d>:

> @@ -50,7 +51,7 @@ public override bool HasActionSets

>              // Light bulb will always invoke this property on the UI thread.

>              AssertIsForeground();

>

> -            if (_actionSets == null)

> +            if (_actionSets.IsDefault)

Got it. I wasn't aware of the explicit operator.

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2froslyn%2fpull%2f4550%2ffiles%23r37085320&data=01%7c01%7cHeejae.Chang%40microsoft.com%7caf7b1b89ec0b4316707208d2a4ba15be%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=CoSLB6AeQ4jCCaARi4fIZZoQfgbGSnJoZgBc0qZ2ijY%3d>.